vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation/proxy/validate_proxy.go

```diff
@@ -0,0 +1,286 @@
+)
+
+const (
+	PluginName = "config.openshift.io/ValidateProxy"
+	maxRetries = 3
+)
+
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return customresourcevalidation.NewValidator(
+			map[schema.GroupResource]bool{
+				configv1.Resource("proxies"): true,
+			},
+			map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+				configv1.GroupVersion.WithKind("Proxy"): ProxyV1{},
+			})
+	})
+}
+
+func toProxyV1(uncastObj runtime.Object) (*configv1.Proxy, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	obj, ok := uncastObj.(*configv1.Proxy)
+	if !ok {
+		return nil, append(allErrs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"Proxy"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"config.openshift.io/v1"}))
+	}
+
+	return obj, nil
+}
+
+type proxyV1 struct {}
+
+func validateProxySpec(spec configv1.ProxySpec) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(spec.HTTPProxy) == 0 && len(spec.HTTPSProxy) == 0  {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec.HTTPProxy"), spec.HTTPProxy, "must be set"))
+	}
+	if len(spec.HTTPProxy) != 0 {
+		if err := validateURI(spec.HTTPProxy); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("HTTPProxy"), spec.HTTPProxy, err.Error()))
+		}
+	}
+	if len(spec.HTTPSProxy) != 0 {
+		if err := validateURI(spec.HTTPSProxy); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("HTTPSProxy"), spec.HTTPSProxy, err.Error()))
+		}
+	}
+	if len(spec.ReadinessEndpoints) != 0 {
+		for _, endpoint := range spec.ReadinessEndpoints {
+			if err := validateURI(endpoint); err != nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("ReadinessEndpoints"), spec.ReadinessEndpoints, err.Error()))
+			} else {
+				if err := validateReadinessEndpoint(endpoint); err != nil {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("spec.ReadinessEndpoints"), spec.ReadinessEndpoints, err.Error()))
+				}
+			}
+		}
+	}
+	if len(spec.NoProxy) != 0 {
+		for _, v := range strings.Split(spec.NoProxy, ",") {
+			v = strings.TrimSpace(v)
+			errDomain := validateDomainName(v, false)
+			_, _, errCIDR := net.ParseCIDR(v)
+			if errDomain != nil && errCIDR != nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("NoProxy"), v, "must be a CIDR or domain, without wildcard characters and without leading or trailing dots ('.')"))
+			}
+			noProxy, err := createNoProxy(installConfig, network)
+		}
+	}
+
+	return allErrs
+}
+
+func (proxyV1) ValidateCreate(uncastObj runtime.Object) field.ErrorList {
+	obj, allErrs := toProxyV1(uncastObj)
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validateProxySpec(obj.Spec)...)
```

Comment:
will we get an operator which writes the status? If yes, we have to enable the status subresource in the CRD. If no, we have to check that status is empty here.

Reply:
similarly for update below

Reply:
yes status will be written by a controller.